### PR TITLE
Filter out terms with empty string value in termlist

### PR DESCRIPTION
**Pull request by JeroenDeDauw**
This is for discussion~~; do not merge.~~

https://groups.google.com/a/wikimedia.de/d/topic/wikimedia-de-tech/NTb6EyOVMCE/discussion

[Bug: T121433](https://phabricator.wikimedia.org/T121433)

### CODE REVIEW

**Comment by thiemowmde** [comment]
I like this very much. I believe this makes the life of all DataModel users easier.

File: src/Term/TermList.php [hunk 1]
```diff
@@ -116,7 +116,12 @@ private function assertIsLanguageCode( $languageCode ) {
 	}
 
 	public function setTerm( Term $term ) {
```
Comment (thiemowmde) [inline src/Term/TermList.php]: Please describe the new behaviour in the documentation of the constructor and this method.

**Comment by Benestar** [comment]
I really like that implementation of my proposal and think it makes sense to do it this way. :+1:

**Comment by brightbyte** [comment]
+2 from me. Not merged, since we haven't decided yet which pull requests should go into the next release.

**Pull request merged**